x-pack/plugins/ml/common/types/anomaly_detection_jobs/job.ts

```diff
@@ -86,3 +87,8 @@ export interface CustomRule {
   scope?: object;
   conditions: any[];
 }
+
+export interface PerPartitionCategorization {
+  enabled: boolean;
+  stop_on_warn?: boolean;
+}
```

Comment:
nit: would be better to extract this type with `TypeOf<typeof analysisConfigSchema['per_partition_categorization']>`

Reply:
i agree that it would be best to have a single source of truth for these definitions. basing these interfaces on our schemas should really be done in a separate PR